### PR TITLE
[wrangler] Deprecate the init command

### PR DESCRIPTION
**Pull request by RamIdeas**
Deprecate the init command and run C3 where we can reliably fallback

~~This should be merged in v3 only~~ No longer a breaking change with —delegate-c3 defaulting to false (thanks 
@penalosa and @petebacondarwin!)

The entire command will be deprecated.

Only part of it's functionality is being delegated to C3 — when the project doesn't exist on disk yet.

If the project does exist (a wrangler.toml is found in the creationDirectory), init's functionality is preserved. We will delegate this functionality to another command within wrangler in the future — before v4 when the init command is removed.

~~Tests still TODO~~

### CODE REVIEW

File: packages/wrangler/src/init.ts [hunk 1]
```diff
@@ -58,6 +59,12 @@ export function initOptions(yargs: CommonYargsArgv) {
 				"The name of the Worker you wish to download from the Cloudflare dashboard for local development.",
 			type: "string",
 			requiresArg: true,
+		})
+		.option("do-not-delegate", {
```
Comment (IgorMinar) [inline packages/wrangler/src/init.ts:63]: could we name this flag without a negation in the name? negation often ends up being used with double negation, which confuses many people. I find that positive flag names are usually more clear and easier to use.

File: packages/wrangler/src/init.ts [hunk 2]
```diff
@@ -185,6 +196,28 @@ export async function initHandler(args: InitArgs) {
...
 		}
+
+		// Deprecate the `init --from-dash` command
+		const replacementC3Command = `\`${packageManager.type} create cloudflare --template pre-existing --name ${fromDashScriptName}\``;
+		logger.warn(
+			`The \`init --from-dash\` command is no longer supported. Please use ${replacementC3Command} instead.\nThe \`init\` command will be removed in a future version.`
+		);
+
+		// C3 will run wrangler with the --do-not-delegate flag to communicate with the API
+		if (!args.doNotDelegate) {
```
Comment (IgorMinar) [inline packages/wrangler/src/init.ts:207]: look, double negation ;-)

**Review by IgorMinar — COMMENTED** [review]
Could you please clarify if we are actually deprecating the init command or only deprecating a significant part of its current implementation and delegating to C3 instead?

I love the use of flags btw and the fact that we can land this as a non-breaking change and in v3 just flip the flag.

My hat tip to @penalosa for suggesting this approach!

**Comment by RamIdeas** [comment]
> Could you please clarify if we are actually deprecating the init command or only deprecating a significant part of its current implementation and delegating to C3 instead?

The entire command will be deprecated.

Only part of it's functionality is being delegated to C3 — when the project doesn't exist on disk yet.

If the project does exist (a wrangler.toml is found in the creationDirectory), init's functionality is preserved. We will delegate this functionality to another command within wrangler in the future — before v4 when the init command is removed.

**Comment by lrapoport-cf** [comment]
@RamIdeas just to clarify, with this change, does C3 take on the functionality currently supported by `init --from-dash`?

**Review by JacobMGEvans — APPROVED** [review]

**Pull request merged**